src/compiler/tracing.ts

```diff
@@ -58,13 +71,14 @@ namespace ts.tracing {
     }
 
     /** Stops tracing for the in-progress project and dumps the type catalog (unless the `fs` module is unavailable). */
-    export function stopTracing(typeCatalog: readonly Type[]) {
+    export function stopTracing(typeCatalog?: readonly Type[]) {
         if (!traceFd) {
             Debug.assert(!fs, "Tracing is not in progress");
             return;
         }
 
         Debug.assert(fs);
+        Debug.assert(!!typeCatalog === (mode !== Mode.Server));

```

Comment:
This looks confusing to me more than `mode === Mode.Build`...  (But same confusion here)

Reply:
Since `Project` is a valid value, that condition would not be equivalent.  I intended this to be read as `typeCatalog` if and only if `mode !== Mode.Server`.

Reply:
Add a comment.